sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CallerInformationParametersShouldBeLast.CSharp9.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        public void OuterMethod() {
+            void Local1([CallerFilePath] string callerFilePath = "") { }
+            void Local2(string other, [CallerFilePath] string callerFilePath = "") { }
+            void Local3([CallerFilePath] string callerFilePath = "", string other = "") { } // FN
```

Comment:
I think we could also add a test case with local functions in the context of top-level functions.